client/client.go

```diff
@@ -5,21 +5,58 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/deepmap/oapi-codegen/pkg/securityprovider"
 	"github.com/hashicorp/go-cleanhttp"
 	"github.com/hashicorp/go-retryablehttp"
 	"github.com/pkg/errors"
+	"golang.org/x/time/rate"
 )
 
+// last retry will wait 64 seconds (2^6)
+// will also listen to 'Retry-After' header
+
+const (
+	rpm              = 600
+	bucketsPerMinute = 60
+	maxRetries       = 6
+	minRetryWait     = 1 * time.Second
+	maxRetryWait     = 65 * time.Second
+)
+
+type RateLimitedClient struct {
+	client      *http.Client
+	rateLimiter *rate.Limiter
+}
+
+func (c *RateLimitedClient) Do(req *http.Request) (*http.Response, error) {
+	if c.rateLimiter == nil {
+		return nil, errors.New("rate limiter not set")
+	}
+	ctx := context.Background()
```

Comment:
You never want to use context.Background() unless you're setting up an entrypoint to a binary, otherwise you lose all control over timeouts etc.

http.Requests have a Context() method on them that provides you the current context.

Reply:
done